pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -259,8 +270,9 @@ func (c *ReverseExpandQuery) execute(
 		return err
 	}
 
-	subg, subgctx := errgroup.WithContext(ctx)
-	subg.SetLimit(int(c.resolveNodeBreadthLimit))
+	pool := pool.New().WithContext(ctx)
```

Comment:
The sourcegraph concurrency pool works like I would have expected `errgroup` to behave. What I noticed is that when ctx is cancelled, the errgroup still waits for the goroutines to return, and thus it can hang. However, using the sourcegraph concurrency pool the pool is immediately interrupted and returns promptly.

I made this change because a) it was necessary to get the concurrency behavior that I expected when context is cancelled on group of goroutines, and b) it minimized the complexity of the change as compared to introducing waitgroups etc..

Reply:
> using the sourcegraph concurrency pool the pool is immediately interrupted and returns promptly.

does this mean that no more elements will be sent through the channel? 

if so, why don't we `close(resultsChan)` in all cases?